src/index.ts

```diff
@@ -505,27 +507,52 @@ export function create (rawOptions: CreateOptions = {}): Register {
       const service = ts.createLanguageService(serviceHost, registry)
 
       const updateMemoryCache = (contents: string, fileName: string) => {
+        let shouldIncrementProjectVersion = false
         const fileVersion = fileVersions.get(fileName) || 0
+        const isFileInCache = fileVersion !== 0
 
         // Add to `rootFiles` when discovered for the first time.
-        if (fileVersion === 0) rootFileNames.push(fileName)
+        if (!isFileInCache) {
+          rootFileNames.push(fileName)
+          // Modifying rootFileNames means a project change
+          shouldIncrementProjectVersion = true
```

Comment:
Nit: You could just do `projectVersion++` here - it's essentially the same, an extra +1 every now and then isn't a big deal.